### DISCUSSION TITLE
Stargate V2 Integration Test status

### DISCUSSION THREAD

**Discussion by tatu-at-datastax** [stargate/stargate#1516] [Show and tell]
(note: description modified regularly with updates -- last update 2022-02-08)

Due to changes between Stargate V1 and V2 (rearchitecting), existing integration test suite mostly failed initially.
Much of the initial upgrade work initially was focused on getting `RestApiv2Test` tests to pass.
But there were a few other integration tests with failures; this document tracks the state of Integration Testing.

Some of the remaining failures are due Stargate V2 not supporting "rest v1" api (but only current "rest v2" -- support for "rest v1" may or may not be added in future).

Status as of 2022-02-08:

* ALL GREEN!
* Occasional regressions due to changes in StargateV1 (would be good to figure out automation workflow detecting regression)

### RestApiJWTAuthTest (0 fails)

This class includes cases against both "rest v1" api and "rest v2" api. The "rest v1" cases fail as expected. A portion of the "rest v2" APIs are passing (6 tests).
Beyond this, any of the cases that involve row-level access control (RLAC) would fail as RLAC is not implemented for SGv2/REST: these tests (5+5) are ignored.

* Pass rate when run from IDE: 6/6; 16 ignored
* Pass rate from CI:  same as above

### RestApiMetricsTest (0 fails)

* Pass rate when run from IDE: 3/3
* Pass rate from CI:  same as above

This end point defines 2 main tests, one with 2 variations. Original failure was due to wrong port: REST metrics will now be co-located with REST API instead of being centralized and running on different port.

### RestApiTest (all ignored)

Since this test runs against "v1" of Stargate REST API -- which Stargate V2 REST API does not currently support (and may or may not every support) -- tests are `@Ignore`d (as of 12-Jan-2022).

### RestApiV2RowsTest (0 fails)

These tests run against "v2" of Stargate REST API, fully supported by Stargate V2.

* Pass rate: 62/62

### RestApiV2SchemaTest (0 fails)

These tests run against "v2" of Stargate REST API, fully supported by Stargate V2.

* Pass rate: 58/58

### RestApiv2DseTest (0 fails)

Has a set of tests specific to DSE backend, specifically Storage-Attached Index (SAI) handling.
Functionality is invoked using regular RESTv2 API but will only pass with DSE backend.

Note: does not run along with CI runs other than DSE.

### RestHealthCheckerTest (0 fails)

These tests were rewritten to only cover minimal liveness and readiness endpoints: more advanced readiness checks that checked multiple APIs rely on OSGI and will not be migrated as-is to SGv2 (since we are splitting up APIs into different services).

### GRPC: SchemaTest (0 fails)

No fails.

**Comment by jeffreyscarpenter** [comment]
Should `RestHealthCheckerTest` be included in this list?

**Reply by tatu-at-datastax** [reply]
Yes, will add.

**Comment by tatu-at-datastax** [comment]
All Green, finally!
